src/Integration.Vsix.UnitTests/Analysis/MuteIssueCommandTests.cs

```diff
@@ -0,0 +1,277 @@
+        activeSolutionBoundTrackerMock
+            .SetupGet(x => x.CurrentConfiguration)
+            .Returns(ConnectedModeBinding);
+        
+        ThreadHelper.SetCurrentThreadAsUIThread();
+        var oleStatus = testSubject.OleStatus;
+
+        oleStatus.Should().Be(InvisibleAndDisabled);
+        testSubject.Visible.Should().BeFalse();
+        testSubject.Enabled.Should().BeFalse();
+    }
+    
+    [TestMethod] 
+    public void QueryStatus_NotInConnectedMode_VisibleButDisabled()
+    {
+        var testSubject = CreateTestSubject(out var errorListHelperMock, out _, out _, out var activeSolutionBoundTrackerMock, out _, out _);
+        SonarCompositeRuleId.TryParse("javascript:S333", out SonarCompositeRuleId ruleId);
+        errorListHelperMock.Setup(x => x.TryGetRuleIdFromSelectedRow(out ruleId)).Returns(true);
+        activeSolutionBoundTrackerMock
+            .SetupGet(x => x.CurrentConfiguration)
+            .Returns(BindingConfiguration.Standalone);
+        
+        ThreadHelper.SetCurrentThreadAsUIThread();
+        var oleStatus = testSubject.OleStatus;
+
+        oleStatus.Should().Be(VisibleButDisabled);
+        testSubject.Visible.Should().BeTrue();
+        testSubject.Enabled.Should().BeFalse();
+    }
+    
+    [TestMethod] 
+    [DataRow("cpp:S111")]
+    [DataRow("c:S222")]
+    [DataRow("javascript:S333")]
+    [DataRow("typescript:S444")]
+    [DataRow("secrets:S555")]
+    [DataRow("css:S555")]
+    public void QueryStatus_ConnectedModeAndSupportedIssue_VisibleAndEnabled(string errorCode)
+    {
+        var testSubject = CreateTestSubject(out var errorListHelperMock, out _, out _, out var activeSolutionBoundTrackerMock, out _, out _);
+        SonarCompositeRuleId.TryParse(errorCode, out SonarCompositeRuleId ruleId);
+        errorListHelperMock.Setup(x => x.TryGetRuleIdFromSelectedRow(out ruleId)).Returns(true);
+        activeSolutionBoundTrackerMock
+            .SetupGet(x => x.CurrentConfiguration)
+            .Returns(ConnectedModeBinding);
+        
+        ThreadHelper.SetCurrentThreadAsUIThread();
+        var oleStatus = testSubject.OleStatus;
+
+        oleStatus.Should().Be(VisibleAndEnabled);
+        testSubject.Visible.Should().BeTrue();
+        testSubject.Enabled.Should().BeTrue();
+    }
+    
+    [TestMethod]
+    public void Execute_DoesNothingWhenIssueCannotBeSelected()
+    {
+        // todo roslyn issues should be checked in this case, but it's not yet implemented
+        var testSubject = CreateTestSubject(out var errorListHelperMock, out _, out _, out _, out _, out _);
+        var issue = Mock.Of<IFilterableIssue>();
+        errorListHelperMock.Setup(x => x.TryGetIssueFromSelectedRow(out issue)).Returns(false);
+
+        testSubject.Invoke();
+        
+        errorListHelperMock.Verify(x => x.TryGetIssueFromSelectedRow(out issue), Times.Once);
+    }
+    
+    [TestMethod]
+    public void Execute_DoesNotFindServerIssue_TODO()
```

Comment:
Do we need this test? I don't believe it should be responsibility of the command.

Reply:
Well, the command is the entity that has the knowledge that the issue was not found. As to what it's supposed to do with that info, there's a todo in the test to add that check